tokio-macros/src/lib.rs

```diff
@@ -56,17 +55,15 @@ pub fn main(args: TokenStream, item: TokenStream) -> TokenStream {
     let attrs = &input.attrs;
 
     if input.asyncness.is_none() {
-        let tokens = quote_spanned! { input.span() =>
-          compile_error!("the async keyword is missing from the function declaration");
-        };
-
-        return TokenStream::from(tokens);
+        let msg = "the async keyword is missing from the function declaration";
+        return syn::Error::new_spanned(input.decl.fn_token, msg)
```

Comment:
There is no change (perhaps the clearest is to highlight `fn`.).
```text
error: the async keyword is missing from the function declaration
  --> tokio\examples\echo.rs:34:1
   |
34 | fn main() -> Result<(), Box<dyn Error>> {
   | ^^
```

Reply:
Note that the original version highlights the entire function when procmacro2_semver_exempt is active.